File: src/router/routes/getsingleimage.go
package routes

import (
	"encoding/json"
	"net/http"
	"router/sqldb"
)

//GetSingleImageHandler POST API
func GetSingleImageHandler(res http.ResponseWriter,req *http.Request,){
	SetHeaders(&res)
	response := make(map[string]string)

	if(sqldb.DB==nil){
		response["ModalTitle"]="Service Unavailable...";
		response["ModalBody"]="Signup Service is unavailable right now... Please try again later";
		res.WriteHeader(http.StatusServiceUnavailable)
		panic("503 Error")
	}

	defer func(){
		if r := recover(); r!=nil{
			json.NewEncoder(res).Encode(response);
		}
	}()

	req.ParseMultipartForm(0)
	email  := req.Form.Get("email");
	token := req.Form.Get("token");


	if(!IsValidUser(email,token)){
        response["ModalTitle"]="Not Authorized...";
        response["ModalBody"]="You are not authorized...";    
        res.WriteHeader(http.StatusUnauthorized)
		panic("401 Error");
	}


	filename := req.Form.Get("filename")

	stmt := "select imagedata from sureify.images where email = '"+email+"' and filename='"+filename+"';";
	rows,_ := sqldb.DB.Query(stmt)
	if(rows.Next()){
		imagedata := ""
		rows.Scan(&imagedata)
		response["imagedata"] = imagedata;
		response["fromgo"] = "true"
	}
	json.NewEncoder(res).Encode(response)
}

File: api/getsingleimage.php
<?php 
  
  require("./checkvaliduserhandler.php");
  $response = array();
  $conn = null;
  
  try {

    $email = verifyUserHandler("Fetch Single Image",$response,$conn);
    $filename = $_POST["filename"];
    
    $sql = "select imagedata from images where email = '$email' and filename='$filename';";
    $res = $conn->query($sql); 
    
    if($res->num_rows>0){
      $row = $res->fetch_assoc();
      $imageData = $row["imagedata"];
      $response["imagedata"] = $imageData;
    }
    
    echo json_encode($response); 

  }catch(Exception $exception){
    echo json_encode($response);
}
?>
